sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -42,7 +42,21 @@ public class EventData
         /// <seealso cref="BinaryData" />
         /// <seealso cref="EventData.Properties" />
         ///
-        public BinaryData EventBody => _amqpMessage.GetEventBody();
+        public BinaryData EventBody
+        {
+            get => _amqpMessage.GetEventBody();
+            set => _amqpMessage.Body = AmqpMessageBody.FromData(MessageBody.FromReadOnlyMemorySegment(value.ToMemory()));
+        }
+
+        /// <summary>
+        /// <inheritdoc cref="EventBody"/>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override BinaryData Data
```

Comment:
Trying out an interface instead - only one interface is needed because we can implement the properties explicitly and throw on the ReceivedMessage setters - thanks to @pakrym for the idea.